src/rootfs.rs

```diff
@@ -68,17 +68,15 @@ pub fn prepare_rootfs(spec: &Spec, rootfs: &Path, bind_devices: bool) -> Result<
         }
     }
 
-    let olddir = getcwd()?;
-    chdir(rootfs)?;
+    chdir(rootfs).context("failed to set current directory to rootfs")?;
```

Comment:
Usually I don't like chdir inside one function since it has side effects, making it much harder for ourselves in the future. I prefer to either chdir back, leaving the cwd the same after the function returns, or not chdir at all. In this case, I think the right thing to do is to remove chdir logic all together here. Instead, we make the `setup_default_symlinks` work with rootfs.